File: packages/now-python-wsgi/now_handler.py
import base64
import json
import requests
import wsgiadapter
import __NOW_HANDLER_FILENAME as app

session = requests.Session()

try:
    app = app.api
    session = app.api.requests
except AttributeError:
    try:
        app = app.app
        session.mount("http://;/", wsgiadapter.WSGIAdapter(app))
    except AttributeError:
        app = app.application
        session.mount("http://;/", wsgiadapter.WSGIAdapter(app))


def now_handler(event, context):
    payload = json.loads(event["body"])
    path = payload["path"]
    headers = payload["headers"]
    method = payload["method"]
    encoding = payload.get("encoding")
    body = payload.get("body")

    if (
        (body is not None and len(body) > 0) and
        (encoding is not None and encoding == "base64")
    ):
        body = base64.b64decode(body)

    res = session.request(method, "http://;" + path, headers=headers, data=body)

    return {
        "statusCode": res.status_code,
        "headers": dict(res.headers),
        "body": res.text,
    }

File: packages/now-ruby/now_handler.rb
require 'webrick'
require 'json'

require_relative _NOW_HANDLER_FILE

module NowHandler
  # server = WEBrick::HTTPServer.new :Port => 3000
  # server.mount '/', Handler
  # trap('INT') { server.stop }
  # server.start


  def self.now_handler(event:, context:)
    server = WEBrick::HTTPServer.new :Port => 3000
    Thread.new(server) do |server|
      if Object.const_defined?('Handler')
        server.mount '/', Handler
      else
        server.mount_proc "/" do |req, res|
        payload = JSON.parse event['body']
        path = payload['path']
        headers = payload['headers']
        http_method = payload['method']

        res.body = "#{http_method}, 'http://0.0.0.0:3000'#{path} #{headers}"
        res.status = 200
        res.headers = headers
      end
      end
    end
    trap('INT') { server.stop }
    server.start
  end
end